### PR TITLE
chore: update conditions on existing preference check

### PR DESCRIPTION
**Pull request by cellomatt**
### Description

This PR updates the code examples on the React preference center example to add a conditional check on whether the `preferenceKey` exists. This should resolve the TS errors reported by a customer [here](https://github.com/knocklabs/javascript/issues/153).

### CODE REVIEW

**Comment by JEverhart383** [comment]
Just quick summary of what I did here:

1. added some checks to avoid that type error if the project is copied into TS
2. updated the `PrefereceCenter` component in the Notion feed example to use full TS
3. Linked that in the opening paragraph if people want a full TS example

After chatting with @cellomatt we didn't think we should convert this whole example to TS since it would be the only TS example in the docs (at least that we could see)

File: content/in-app-ui/react/preferences.mdx [hunk 1]
```diff
@@ -283,13 +306,21 @@ export default function PreferenceCenter() {
       ...localPreferences,
     };
 
-    //Here we'll make updates to the preference set based on the preferenceType
+    // Here we'll make updates to the preference set based on the preferenceType
     // and override existing channelTypeSettings
-    if (preferenceType === "category") {
+    // since Workflow and Category preferences can also be a Boolean,
+    // we'll check if the preferenceKey contains an channel_types object
```
Comment (cellomatt) [inline content/in-app-ui/react/preferences.mdx:312]:
```suggestion
    // we'll check if the preferenceKey contains a channel_types object
```

File: content/in-app-ui/react/preferences.mdx [hunk 2]
```diff
@@ -324,7 +357,9 @@ export default function PreferenceCenter() {
             preferenceType="workflow"
             preferenceKey={workflow}
             channelTypeSettings={
-              localPreferences?.workflows[workflow]?.channel_types
+              typeof localPreferences.categories[category] === "object"
+                ? localPreferences?.categories[category]?.channel_types
```
Comment (cellomatt) [inline content/in-app-ui/react/preferences.mdx:361]:
```suggestion
              typeof localPreferences.workflows[workflow] === "object"
                ? localPreferences?.workflows[workflow]?.channel_types
```
I think?

File: content/in-app-ui/react/preferences.mdx [hunk 3]
```diff
@@ -369,13 +404,21 @@ const onPreferenceChange = async ({
...
+    preferenceType === "category" &&
+    typeof preferenceUpdate.categories[preferenceKey] === "object"
+  ) {
     preferenceUpdate.categories[preferenceKey].channel_types =
       channelTypeSettings;
   }
-  if (preferenceType === "workflow") {
+  if (
+    preferenceType === "workflow" &&
+    typeof preferenceUpdate.categories[preferenceKey] === "object"
```
Comment (cellomatt) [inline content/in-app-ui/react/preferences.mdx:420]:
```suggestion
    typeof preferenceUpdate.workflow[preferenceKey] === "object"
```

File: content/in-app-ui/react/preferences.mdx [hunk 4]
```diff
@@ -554,13 +625,21 @@ export default function PreferenceCenter() {
...
+      preferenceType === "category" &&
+      typeof preferenceUpdate.categories[preferenceKey] === "object"
+    ) {
       preferenceUpdate.categories[preferenceKey].channel_types =
         channelTypeSettings;
     }
-    if (preferenceType === "workflow") {
+    if (
+      preferenceType === "workflow" &&
+      typeof preferenceUpdate.categories[preferenceKey] === "object"
```
Comment (cellomatt) [inline content/in-app-ui/react/preferences.mdx:641]:
```suggestion
      typeof preferenceUpdate.workflow[preferenceKey] === "object"
```

**Review by JEverhart383 — APPROVED** [review]
Had @cellomatt review this since I contributed

**Pull request merged**